docs/source/advanced/training_tricks.rst

```diff
@@ -251,8 +256,8 @@ with the parameters of the algorithm. This can be done by invoking the
     # update hparams of the model
     model.hparams.lr = new_lr
 
-    # Fit model
-    trainer.fit(model)
+    # Fit model, with a datamodule (assuming it's defined before)
```

Comment:
this code is just for representing how it works, and won't work anyways even without a datamodule.
I'd say instantiate datamodule too like:
```py
model = MyModelClass(hparams)
datamodule = MyDataModuleClass(hparams)
```
also, you might need to fix the complete example since it doesn't align properly.

Reply:
Thanks, @rohitgr7 for the comment - don't want to define the `datamodule` here, since the data modules are discussed towards the end of the page (in the documentation). However, I agree we can fix the examples but let's keep it for another PR. Since this PR intends to give a hint that passing a data module is valid and also links to the doc for `tune` API. Let me know if you disagree.

Reply:
Post offline discussion with @rohitgr7 - we agreed that we want to extend the note showing the usage of `datamodule`. Also, there shouldn't be a need to fix the calls for each instance of `trainer.fit(model)/trainer.tune(model)` to `trainer.fit(model, datamodule)/trainer.tune(model, datamodule)`.

The PR has been updated. Thanks, @rohitgr7. Also, thanks @kaushikb11 for the question :)